ext/AdvancedHMCADTypesExt.jl

```diff
@@ -0,0 +1,41 @@
+module AdvancedHMCADTypesExt
+
+if isdefined(Base, :get_extension)
+    using AdvancedHMC:
+        AbstractMetric,
+        Hamiltonian,
+        LogDensityModel,
+        LogDensityProblems,
```

Comment:
The extension should depend on LogDensityProblems if it loads it, indirect imports should be avoided.